src/BaGet/Entities/PostgreSqlContext.cs

```diff
@@ -0,0 +1,28 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Npgsql;
+
+namespace BaGet.Entities
+{
+    public class PostgreSqlContext : AbstractContext<PostgreSqlContext>
+    {
+        /// <summary>
+        /// The PostgreSql error code for when a unique constraint is violated.
+        /// </summary>
+        private const string UniqueConstraintViolationErrorCode = "23505";
+
+        public PostgreSqlContext(DbContextOptions<PostgreSqlContext> options)
+            : base(options)
+        {
+        }
+
+        public override bool IsUniqueConstraintViolationException(DbUpdateException exception)
+        {
+            return exception.InnerException is PostgresException postgresException &&
+                   postgresException.SqlState.Contains(UniqueConstraintViolationErrorCode);
```

Comment:
`SqlState` is a string representing the error code, right? If so, please make `UniqueConstraintViolationErrorCode ` and use `int.TryParse` here instead of `Contains`.